test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -441,12 +463,14 @@ class ExtAuthzHttpIntegrationTest : public HttpIntegrationTest,
 
     // Send back authorization response with "baz" and "bat" headers.
     // Also add multiple values "append-foo" and "append-bar" for key "x-append-bat".
+    // Also tell Envoy to remove "remove-me" header before sending to upstream.
     Http::TestResponseHeaderMapImpl response_headers{
         {":status", "200"},
         {"baz", "baz"},
         {"bat", "bar"},
         {"x-append-bat", "append-foo"},
         {"x-append-bat", "append-bar"},
+        {"x-envoy-auth-headers-to-remove", "remove-me"},
```

Comment:
Seems like we haven't test it for multiple header names and its variations? 

1. `{"x-envoy-auth-headers-to-remove", "remove-1,remove-1"}`
1. `{"x-envoy-auth-headers-to-remove", "remove-1, remove-2"}`
1. `{"x-envoy-auth-headers-to-remove", "remove-1,,,remove-2"}`

And also when we have multiple entries of "x-envoy-auth-headers-to-remove".

```
{"x-envoy-auth-headers-to-remove", "remove-1,remove-2"}
{"x-envoy-auth-headers-to-remove", "remove-2,remove-3"}
```

Reply:
Probably you can do this in unit tests below.

Reply:
@dio Implemented in a new test.